src/ansys/speos/core/sensor.py

```diff
@@ -3742,3 +3742,617 @@ def set_geometries(self, geometries: [List[GeoRef]]) -> Sensor3DIrradiance:
             gr.to_native_link() for gr in geometries
         ]
         return self
+
+
+class SensorXMPIntensity(BaseSensor):
+    """Class for XMP intensity sensor.
+
+    Parameters
+    ----------
+    project
+    name
+    description
+    metadata
+    sensor_instance
+    default_values
+    """
+
+    def __init__(
+        self,
+        project: project.Project,
+        name: str,
+        description: str = "",
+        metadata: Optional[Mapping[str, str]] = None,
+        sensor_instance: Optional[ProtoScene.SensorInstance] = None,
+        default_values: bool = True,
+    ) -> None:
+        if metadata is None:
+            metadata = {}
+
+        super().__init__(
+            project=project,
+            name=name,
+            description=description,
+            metadata=metadata,
+            sensor_instance=sensor_instance,
+        )
+
+        # Attribute gathering more complex intensity type
+        self._type = None
+        self._layer_type = None
+
+        if default_values:
+            # Default values template
+            self.set_type_photometric()
+            self.set_orientation_x_as_meridian()
+            self.set_viewing_direction_from_source()
+            # Default values properties
+            self.set_axis_system().set_layer_type_none()
+
+    @property
+    def nearfield(self) -> bool:
+        """Property containing if the sensor is positioned in nearfield or infinity."""
+        if self._sensor_template.intensity_sensor_template.HasField("near_field"):
+            return True
+        else:
+            return False
+
+    @nearfield.setter
+    def nearfield(self, value):
+        if value:
+            self._sensor_template.intensity_sensor_template.near_field.SetInParent()
+        else:
+            if self._sensor_template.intensity_sensor_template.HasField("near_field"):
+                self._sensor_template.intensity_sensor_template.ClearField("near_field")
+
+    @property
+    def cell_distance(self):
+        """Distance of the Detector to origin in mm."""
+        if self.nearfield:
+            return self._sensor_template.intensity_sensor_template.near_field.cell_distance
+        else:
+            return None
+
+    @cell_distance.setter
+    def cell_distance(self, value):
+        if self.nearfield:
+            self._sensor_template.intensity_sensor_template.near_field.cell_distance = value
+        else:
+            raise TypeError("Sensor position is not in nearfield")
+
+    @property
+    def cell_integration_angle(self):
```

Comment:
integration angle is only available in the polar intensity sensor

Reply:
in correct it is availble on nearfield

Reply:
![image](https://github.com/user-attachments/assets/9324040d-5c98-4e4f-883f-bf6950315aed)
it is just defined as diameter
shall i set the diameter and compute the angle?